docs/resources/deployment.md

```diff
@@ -0,0 +1,67 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "juju_deployment Resource - terraform-provider-juju"
+subcategory: ""
+description: |-
+  A resource that represents a Juju deployment.
+---
+
+# juju_deployment (Resource)
+
+A resource that represents a Juju deployment.
+
+## Example Usage
+
+```terraform
+resource "juju_deployment" "this" {
+  name = "foobar" # optional, set to charm name when absent
+
+  model = juju_model.development.uuid # required, model uuid
+
+  charm {
+    name     = "hello-kubecon" # required, supports CharmHub charms only
+    revision = ""              # optional, default: -1
+    channel  = ""              # optional, default: stable
```

Comment:
For channel, we have to follow the track/risk/branch definition from [here](https://juju.is/docs/olm/deploy-a-charm-from-charmhub#heading--specify-a-charmed-operator-channel)

Reply:
By default it can be empty so Juju can determine which is the default charm.

Reply:
@juanmanuel-tirado Ok, I misunderstood slightly. Thanks for the documentation link. 

Would this be more correct way of addressing a Charm (along with defaults)?

```hcl
  charm {
    name     = "hello-kubecon"            # required, supports CharmHub charms only
    channel  = "edge"                     # optional, specified as <track>/<risk>/<branch>, default: stable
    revision = 14                         # optional, default: -1
    series   = "trusty"                   # optional
  }
```

Reply:
Yes that is. Bear in mind that the tuple channel-revision may not always make sense. For example, revision 13 could be stable instead of edge. Something similar can happen with series. Checking the correctness of these values is something to be done by Juju so we do not need to address it here.